lib/matplotlib/patches.py

```diff
@@ -479,11 +480,15 @@ def get_hatch(self):
         'Return the current hatching pattern'
         return self._hatch
 
-    @artist.allow_rasterization
-    def draw(self, renderer):
-        'Draw the :class:`Patch` to the given *renderer*.'
-        if not self.get_visible():
-            return
+    @contextlib.contextmanager
+    def _prepare_path_drawer(self, renderer):
+        """
+        ``draw()`` helper factored out for sharing with `FancyArrowPatch`.
+
+        Yields a callable that can be called as a bound ``draw_path`` method
```

Comment:
Would this be more clear?

> Yields a ``draw_path``-like function that is bound to the `GraphicsContext` of the *renderer*. Technically, this is a `partial` function of ``draw_path`` with the first argument (the ``GraphicsContext``) already filled in. The arguments of the returned function are the remaining arguements of ``draw_path``.

Reply:
reworded the docstring to something else...